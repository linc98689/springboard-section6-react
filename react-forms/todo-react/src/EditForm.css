.edit-form-shown{
    z-index:2;
}
.edit-form-unshown{
    z-index:-1;
}
.edit-form{
    position:fixed;
    top: 0px;
    left: 50%;
    transform: translate(-50%, 0);
    width: 100vw;
    height: 100vh;
    background-color:rgba(230, 246, 246, 0.95);
    display: flex;
    /* justify-content: center; */
    align-items: center;
    flex-direction: column;
}

.edit-form-title{
    padding-top:100px;
}
.edit-buttons{
    margin-top:10px;
    padding-left: 3px 6px;
}
.edit-input{
    display:block;
    width:80vw;
    padding: 3px 10px;
    border: 1px solid rgb(3, 99, 154);
}
input:focus{
    outline: 2px solid rgb(3, 99, 154);
    outline-offset: 1px;
    border:1px solid white;
}
.edit-form-warning{
    margin:1px;
    padding:0;
    font-size:0.7em;
    color:rgb(246, 14, 14);
}
