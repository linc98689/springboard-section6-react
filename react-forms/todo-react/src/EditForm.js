import React, {useState, useRef, useEffect} from "react";
import "./EditForm.css";

const EditForm = ({task, updateTask, updateCurrentTaskIdx, className_2}) =>{
    const [formData, setFormData] = useState("");
    useEffect(()=>{
        setFormData(task);
        refInput.current.focus();
    }, [task]);

    const refInput = useRef();
    const refTaskWarning = useRef();
    const handleChange = (evt) =>{
        const task = evt.target.value;
        refTaskWarning.current.style.opacity = (task===""? 1:0);
        setFormData(data => task);
    };

    const handleSubmit = (evt) =>{
        evt.preventDefault();
        if(formData==="")
            return;
        const tk= formData;
        updateTask(tk);
        updateCurrentTaskIdx(-1);
    }

    const handleCancel = (evt)=>{
        updateCurrentTaskIdx(-1);
    };

    return (
        <div className={`edit-form ${className_2}`}>
            <h2 className="edit-form-title"> Editing a task</h2>
            <form>
                <input className="edit-input" id="edit-task" type="text" value={formData}
                onChange={handleChange} ref={refInput} autoFocus/> 
                <p className="newform-warning" 
                style={{opacity:0}} ref={refTaskWarning}>required</p>
                <div className="edit-buttons">
                    <button type="button" onClick={handleCancel}>Cancel</button>
                    <button type="submit" onClick={handleSubmit}>Save</button>
                </div>
            </form>
        </div>
        );
    }

export default EditForm;