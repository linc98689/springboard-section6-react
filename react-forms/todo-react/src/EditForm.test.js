import { render} from '@testing-library/react';
import '@testing-library/jest-dom';
import EditForm from "./EditForm";

// smoke test
test("should render", ()=>{
    render(<EditForm  task="wash car"/>);
});

// snapshot test
test("should not change", ()=>{
    const {asFragment} = render(<EditForm task="wash car" />);
    expect(asFragment()).toMatchSnapshot();
});