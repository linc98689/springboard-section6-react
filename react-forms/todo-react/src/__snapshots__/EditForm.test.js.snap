// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change 1`] = `
<DocumentFragment>
  <div
    class="edit-form undefined"
  >
    <h2
      class="edit-form-title"
    >
       Editing a task
    </h2>
    <form>
      <input
        class="edit-input"
        id="edit-task"
        type="text"
        value="wash car"
      />
      <div
        class="edit-buttons"
      >
        <button
          type="button"
        >
          Cancel
        </button>
        <button
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
